### PR TITLE
feat(models): add Location Group, Membership, and Invite models

### PR DESCRIPTION
**Pull request by denisvmedia**
## Summary

Phase 1 of Location Groups (#1219) — as the **root of the merge train**, this branch has absorbed all downstream phases (#1223–#1231 plus follow-ups) on the way up. What you're reviewing is the full Location Groups feature rebased onto master.

### What this PR actually contains

**Models (pure additions)**
- **LocationGroup** — tenant-scoped group with non-guessable slug (22+ chars, base64url), icon, active/pending_deletion status. Tenant-only RLS.
- **GroupMembership** — links users to groups with per-group roles (admin/user). Unique on `(tenant_id, group_id, member_user_id)`.
- **GroupInvite** — single-use invite links with configurable expiry (default 24h), created_by / used_by audit trail; `Token` tagged `userinput:"false"`.
- **TenantGroupAwareEntityID** — `{id, uuid, tenant_id, group_id, created_by_user_id}` base for data entities.
- **TenantOnlyEntityID** — `{id, uuid, tenant_id}` base for tenant-scoped metadata (LocationGroup / GroupMembership / GroupInvite) that don't carry a user_id column.
- **GroupAware** / **CreatedByUserAware** interfaces.

**Breaking schema / API changes folded in from downstream phases**
- `users.role` column and the tenant-level `UserRole` concept are removed. Roles are now per-group via `GroupMembership.role`. `users_role_idx` dropped, users-table RLS tightened.
- Data tables (locations, areas, commodities, files, exports, images, invoices, manuals, restore_operations, restore_steps) gain a `group_id` column (nullable in phase 5, backfilled + NOT NULL in phase 6 via a hand-written migration), and their RLS policies move from `(tenant_id, user_id)` to `(tenant_id, group_id)`.
- `user_id` on data tables renamed to `created_by_user_id` (audit-only, not access control).
- The old `/api/v1/users` admin surface is removed — `RequireAdmin` was 403-ing everyone until group-based admin auth is designed (tracked in #1246).

**Registry / API layer**
- `RLSGroupRepository` (parallel to `RLSRepository`) — transaction-scoped `app.current_group_id` via `SET LOCAL`.
- `GroupService` with CreateGroup (compensating delete on membership failure), AcceptInvite (CAS-based `MarkUsed` on the invite, tenant verification), cross-group revocation guard, `CheckGroupMembership` with proper error surfacing.
- `/api/v1/groups` (create/list/update/delete/members/invites) + `/api/v1/invites/{token}` (public GET, auth POST) + group-scoped data routes mounted at `/api/v1/g/{groupSlug}/...`.
- `GroupSlugResolverMiddleware` runs **before** `RegistrySetMiddleware` so per-request registries see the group context.
- Tenant / user / group RLS setter functions now use `set_config(..., true)` (transaction-local) and drop `SECURITY DEFINER`. See `1776521044_rls_context_transaction_local`.

**Frontend**
- `GroupSelector`, `GroupSettingsView`, `NoGroupView`, `GroupCreateView`, `InviteAcceptView` views.
- Axios interceptor rewrites `/api/v1/{locations,areas,commodities,files,exports,settings,upload-slots,uploads,search}/...` → `/api/v1/g/{slug}/...` when a slug is in `localStorage`.
- `groupStore` (Pinia) with per-group membership, current-role getters, persistent `currentGroupSlug`.
- `/admin/users` route + `UserListView` / `UserForm` / `userService` removed (backend endpoints were already 403-only).

**E2E**
- New Playwright specs: `groups.spec.ts` (incl. last-admin protection), `invite-flow.spec.ts`, `group-data-isolation.spec.ts`.
- `setup-stack.ts` thread-safe against previous leaky sessions; URL-rewrite `waitForResponse` predicates updated for the new group-scoped paths.

### Migrations included

Applied in timestamp order:
1. `1776362918_phase1_add_groups.{up,down}.sql` — create group tables + RLS.
2. `1776371737_phase5_add_nullable_group_id.{up,down}.sql` — nullable `group_id` on data tables (Ptah).
3. `1776371738_phase6_backfill_groups.{up,down}.sql` — hand-written: backfill groups/memberships, backfill `group_id`, flip NOT NULL, rename `user_id` → `created_by_user_id`, switch RLS policies, drop `users.role`.
4. `1776504701_add_tenant_group_indexes.{up,down}.sql` — composite `(tenant_id, group_id)` indexes.
5. `1776519929_drop_user_id_from_group_tables.{up,down}.sql` — remove the unused `user_id` column inherited from the old `TenantAwareEntityID` embedding on group tables.
6. `1776521044_rls_context_transaction_local.{up,down}.sql` — hand-written: replace `set_tenant_context` / `set_user_context` / `set_group_context` with `SET LOCAL` semantics and drop `SECURITY DEFINER` (Ptah can't diff function bodies — tracked upstream in [stokaro/ptah#89](https://github.com/stokaro/ptah/issues/89)).

### Follow-ups tracked separately

- #1245 — e2e coverage for the invite accept flow.
- #1246 — group-based tenant-admin authorization model.
- #1247 — run E2E against the built Docker image instead of dev-mode `go run` + vite.
- #1248 — `main_currency` belongs on the group, not the user (currency is a property of the inventory, not of the person browsing it).

## Test plan

- [x] Full `make test-go` + `make test-go-postgres` green.
- [x] `make lint-go` clean (0 issues).
- [x] Frontend `npm run lint` / `npm run test` / `npm run build` green.
- [x] Smoke Test (docker-compose, login → `/api/v1/g/{slug}/locations` → logout → blacklist) green.
- [x] Kind smoke test (docker + goreleaser) green.
- [x] Swagger docs regenerated and in sync.

### CODE REVIEW

File: go/models/group_membership_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	for _, tt := range tests {
+		c.Run(tt.name, func(c *qt.C) {
+			err := tt.role.Validate()
+			if tt.wantErr {
+				c.Assert(err, qt.IsNotNil)
+			} else {
+				c.Assert(err, qt.IsNil)
+			}
```
Comment (denisvmedia) [inline go/models/group_membership_test.go:33]: if/else/switch/case is FORBIDDEN in this project for test assertions. Either split tests to failure and happy paths, or assert to a particular error here (as a corner case - err == nil).

File: go/models/group_membership_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	for _, tt := range tests {
+		c.Run(tt.name, func(c *qt.C) {
+			err := tt.m.ValidateWithContext(ctx)
+			if tt.wantErr {
+				c.Assert(err, qt.IsNotNil)
+			} else {
+				c.Assert(err, qt.IsNil)
+			}
```
Comment (denisvmedia) [inline go/models/group_membership_test.go:111]: ditto

File: go/models/location_group_test.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	for _, tt := range tests {
+		c.Run(tt.name, func(c *qt.C) {
+			err := tt.status.Validate()
+			if tt.wantErr {
+				c.Assert(err, qt.IsNotNil)
+			} else {
+				c.Assert(err, qt.IsNil)
+			}
```
Comment (denisvmedia) [inline go/models/location_group_test.go:33]: ditto

File: go/models/location_group_test.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	for _, tt := range tests {
+		c.Run(tt.name, func(c *qt.C) {
+			err := tt.group.ValidateWithContext(ctx)
+			if tt.wantErr {
+				c.Assert(err, qt.IsNotNil)
+			} else {
+				c.Assert(err, qt.IsNil)
+			}
```
Comment (denisvmedia) [inline go/models/location_group_test.go:131]: ditto

**Comment by denisvmedia** [comment]
All review comments from this PR have been addressed in follow-up PR #1231.

File: go/models/group_membership.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+type GroupMembership struct {
+	//migrator:embedded mode="inline"
+	TenantAwareEntityID
+
+	// GroupID references the location group this membership belongs to.
+	//migrator:schema:field name="group_id" type="TEXT" not_null="true" foreign="location_groups(id)" foreign_key_name="fk_membership_group"
+	GroupID string `json:"group_id" db:"group_id"`
+
+	// MemberUserID references the user who is a member of the group.
+	// Named MemberUserID (not UserID) to avoid collision with TenantAwareEntityID.UserID.
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_membership.go]: Fixed in b6166c3. Introduced `TenantOnlyEntityID` (EntityID + tenant_id, with no-op `GetUserID`/`SetUserID` to satisfy the `ptrTenantUserAware` type constraint in service-mode `RLSRepository`), and switched `GroupMembership` to embed it instead of `TenantAwareEntityID`. Generated a follow-up migration `1776519929_drop_user_id_from_group_tables` via `./scripts/generate-migration.sh` that drops the now-unused `user_id` column + `fk_entity_user` constraint. Per-record provenance lives in `MemberUserID` (the person in the group).

File: go/models/group_invite.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+	GroupID string `json:"group_id" db:"group_id"`
+
+	// Token is a cryptographically random, URL-safe string used in the invite link.
+	//migrator:schema:field name="token" type="TEXT" not_null="true"
+	Token string `json:"token" db:"token"`
+
+	// CreatedBy is the user ID of the admin who generated the invite.
+	//migrator:schema:field name="created_by" type="TEXT" not_null="true" foreign="users(id)" foreign_key_name="fk_invite_created_by"
+	CreatedBy string `json:"created_by" db:"created_by" userinput:"false"`
+
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_invite.go:45]: Fixed in b6166c3 — same `TenantOnlyEntityID` switch as for `GroupMembership`, and the same follow-up migration drops `group_invites.user_id`. Audit provenance lives in `CreatedBy` and `UsedBy`, not in an inherited column.

File: go/models/group_invite.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	//migrator:embedded mode="inline"
+	TenantAwareEntityID
+
+	// GroupID references the group this invite is for.
+	//migrator:schema:field name="group_id" type="TEXT" not_null="true" foreign="location_groups(id)" foreign_key_name="fk_invite_group"
+	GroupID string `json:"group_id" db:"group_id"`
+
+	// Token is a cryptographically random, URL-safe string used in the invite link.
+	//migrator:schema:field name="token" type="TEXT" not_null="true"
+	Token string `json:"token" db:"token"`
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_invite.go]: Already addressed earlier in the merge train. `GroupInvite.Token` in `go/models/group_invite.go:40` carries `userinput:"false"`:\n\n```go\nToken string `json:"token" db:"token" userinput:"false"`\n```\n\nSo `SanitizeUserInput` zeroes any client-supplied token on create/update.

File: go/models/group_aware.go [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+func (*TenantGroupAwareEntityID) Validate() error {
+	return ErrMustUseValidateWithContext
+}
+
+func (i *TenantGroupAwareEntityID) ValidateWithContext(ctx context.Context) error {
+	return validation.ValidateStructWithContext(ctx, i,
+		validation.Field(&i.TenantID, validation.Required),
+		validation.Field(&i.GroupID, validation.Required),
+		validation.Field(&i.CreatedByUserID, validation.Required),
+	)
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_aware.go:94]: Already addressed earlier in the merge train. `TenantGroupAwareEntityID.ValidateWithContext` in `go/models/group_aware.go:89-95` uses `rules.NotEmpty` for all three string IDs (TenantID / GroupID / CreatedByUserID). The new `TenantOnlyEntityID.ValidateWithContext` follows the same convention.

File: go/models/location_group.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+//
+//migrator:schema:rls:enable table="location_groups" comment="Enable RLS for multi-tenant location group isolation"
+//migrator:schema:rls:policy name="location_group_tenant_isolation" table="location_groups" for="ALL" to="inventario_app" using="tenant_id = get_current_tenant_id() AND get_current_tenant_id() IS NOT NULL AND get_current_tenant_id() != ''" with_check="tenant_id = get_current_tenant_id() AND get_current_tenant_id() IS NOT NULL AND get_current_tenant_id() != ''" comment="Ensures location groups are isolated by tenant; group-level access is enforced in application logic via memberships"
+//migrator:schema:rls:policy name="location_group_background_worker_access" table="location_groups" for="ALL" to="inventario_background_worker" using="true" with_check="true" comment="Allows background workers to access all location groups for processing"
+
+//migrator:schema:table name="location_groups"
+type LocationGroup struct {
+	//migrator:embedded mode="inline"
+	TenantAwareEntityID
+
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/location_group.go:49]: Fixed in b6166c3 — same `TenantOnlyEntityID` switch as for `GroupMembership`/`GroupInvite`, and the same follow-up migration drops `location_groups.user_id`. Creator provenance lives in `LocationGroup.CreatedBy` (which I also promoted to a required field in `LocationGroupRegistry.Create` for the same reason you flagged here).

File: go/models/group_membership_test.go (hunk 1 above)
Comment (denisvmedia) [reply go/models/group_membership_test.go]: Already addressed. `go/models/group_membership_test.go` now uses the `c.Assert(err != nil, qt.Equals, tt.wantErr)` pattern (no `if`/`else`/`switch`/`case` in assertions) — `grep -c "if\|switch" go/models/group_membership_test.go` returns 0.

File: go/models/group_membership_test.go (hunk 2 above)
Comment (denisvmedia) [reply go/models/group_membership_test.go]: Same — already addressed (`group_membership_test.go` follows the convention).

File: go/models/location_group_test.go (hunk 3 above)
Comment (denisvmedia) [reply go/models/location_group_test.go]: Already addressed. `go/models/location_group_test.go` also follows the `c.Assert(err != nil, qt.Equals, tt.wantErr)` convention — 0 `if`/`switch` hits.

File: go/models/location_group_test.go (hunk 4 above)
Comment (denisvmedia) [reply go/models/location_group_test.go]: Same — already addressed (`location_group_test.go` follows the convention).

File: go/apiserver/uploads.go [hunk 10]
```diff
@@ -122,9 +122,9 @@ func (api *uploadsAPI) handleImageUpload(w http.ResponseWriter, r *http.Request)
 	// Create file entity instead of image
 	now := time.Now()
 	fileEntity := models.FileEntity{
-		TenantAwareEntityID: models.TenantAwareEntityID{
-			TenantID: user.TenantID,
-			UserID:   user.ID,
+		TenantGroupAwareEntityID: models.TenantGroupAwareEntityID{
+			TenantID:        user.TenantID,
+			CreatedByUserID: user.ID,
 		},
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/uploads.go:136]: Fixed in 689ca6c. `go/apiserver/uploads.go` now imports `appctx` and every one of the 6 FileEntity constructors in the upload handlers sets `GroupID: appctx.GroupIDFromContext(r.Context())`. Under the mounted route `/api/v1/g/{groupSlug}/uploads/...` the slug resolver middleware has already populated the group on the context, so the handler just reads it.

File: go/apiserver/files.go [hunk 11]
```diff
@@ -195,9 +195,9 @@ func (api *filesAPI) createFile(w http.ResponseWriter, r *http.Request) {
 
 	now := time.Now()
 	fileEntity := models.FileEntity{
-		TenantAwareEntityID: models.TenantAwareEntityID{
-			TenantID: user.TenantID,
-			UserID:   user.ID,
+		TenantGroupAwareEntityID: models.TenantGroupAwareEntityID{
+			TenantID:        user.TenantID,
+			CreatedByUserID: user.ID,
 		},
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/files.go:202]: Fixed in 689ca6c. `createFile` in `go/apiserver/files.go` now populates `GroupID` the same way — from `appctx.GroupIDFromContext(r.Context())`, which GroupSlugResolverMiddleware writes before the handler runs.

File: go/models/group_rls_functions.go [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+// Group RLS Functions for Group-Level Database Isolation
+//
+// These functions provide session-based group context management
+// for Row-Level Security policies to work effectively alongside
+// the existing tenant-based and user-based isolation.
+
+// Group context functions for Row-Level Security
+type GroupRLSFunctions struct {
+	// Function to set the current group context in the session
+	//migrator:schema:function name="set_group_context" params="group_id_param TEXT" returns="VOID" language="plpgsql" security="DEFINER" body="BEGIN PERFORM set_config('app.current_group_id', group_id_param, false); END;" comment="Sets the current group context for RLS policies"
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_rls_functions.go]: Fixed in 689ca6c. Updated `set_group_context` in `go/models/group_rls_functions.go` — it now uses `set_config(..., true)` (transaction-local) and no longer carries `SECURITY DEFINER`. Ptah cannot diff function bodies (filed [stokaro/ptah#89](https://github.com/stokaro/ptah/issues/89) on that), so the DB-side update ships as a hand-written follow-up migration `1776521044_group_rls_transaction_local.{up,down}.sql`. The existing tenant/user equivalents (from an earlier migration) have the same issue and should get the same fix in a follow-up PR once they're not frozen by the #1219 train.

File: go/models/group_rls_functions.go [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+	_ int
+
+	// Function to get the current group ID from the session
+	//migrator:schema:function name="get_current_group_id" returns="TEXT" language="plpgsql" volatility="STABLE" body="BEGIN RETURN current_setting('app.current_group_id', true); END;" comment="Gets the current group ID from session for RLS policies"
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_rls_functions.go:24]: Same fix as the sibling comment — transaction-local `set_config(..., true)` + no `SECURITY DEFINER` on `set_group_context`, landed via 689ca6c and the hand-written `1776521044_group_rls_transaction_local` migration. (Same Ptah limitation as above — tracked in [stokaro/ptah#89](https://github.com/stokaro/ptah/issues/89).)

File: go/apiserver/commodities.go [hunk 14]
```diff
@@ -446,9 +446,9 @@ func (api *commoditiesAPI) listInvoices(w http.ResponseWriter, r *http.Request)
 	var invoices []*models.Invoice
 	for _, file := range files {
 		invoice := &models.Invoice{
-			TenantAwareEntityID: models.TenantAwareEntityID{EntityID: models.EntityID{ID: file.ID}, TenantID: "default-tenant"},
-			CommodityID:         commodity.ID,
-			File:                file.File,
+			TenantGroupAwareEntityID: models.TenantGroupAwareEntityID{EntityID: models.EntityID{ID: file.ID}, TenantID: "default-tenant"},
+			CommodityID:              commodity.ID,
+			File:                     file.File,
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/commodities.go]: Fixed in 689ca6c. All 8 File → Invoice/Manual/Image legacy wrappers in `go/apiserver/commodities.go` now copy the source file's `TenantGroupAwareEntityID` wholesale instead of hard-coding `"default-tenant"` + dropping group_id / created_by. Provenance (tenant_id, group_id, created_by_user_id) is preserved end-to-end — future code that relies on those fields for serialization / link-building / validation will see the correct values.

File: go/models/user.go [hunk 15]
```diff
@@ -13,32 +13,13 @@ import (
 )
 
 var (
-	_ validation.Validatable            = (*UserRole)(nil)
 	_ validation.Validatable            = (*User)(nil)
 	_ validation.ValidatableWithContext = (*User)(nil)
 	_ TenantAwareIDable                 = (*User)(nil)
 	_ json.Marshaler                    = (*User)(nil)
 	_ json.Unmarshaler                  = (*User)(nil)
 )
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/user.go:21]: You're right that the PR description was stale. It was written for an earlier shape of the branch where the group models still embedded `TenantAwareEntityID` (and the phrase "pure additions" was accurate under that shape). The later commits on this PR switched `LocationGroup` / `GroupMembership` / `GroupInvite` to `TenantOnlyEntityID` and generated a migration to drop `user_id` from those tables — at which point the description became misleading.

I've updated the PR description to reflect the current state: the group metadata tables are tenant-only (no `user_id`), a migration drops those columns, and `TenantGroupAwareEntityID` is present for the Phase-6 data-model switch (unused in this PR). No more "pure additions" claim.

File: frontend/src/views/exports/ExportImportView.vue [hunk 16]
```diff
@@ -142,8 +143,19 @@ const handleFileUpload = async (files: File[]) => {
...
+    const csrfToken = getCsrfToken()
+    const headers: Record<string, string> = {}
+    if (token) headers['Authorization'] = `Bearer ${token}`
+    if (csrfToken) headers['X-CSRF-Token'] = csrfToken
+
+    const response = await fetch(`/api/v1/g/${groupSlug}/uploads/restores`, {
       method: 'POST',
+      headers,
       body: formData,
     })
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/views/exports/ExportImportView.vue]: Fixed in 8ae91a7. The upload now goes through the shared axios client (`api.post('/api/v1/uploads/restores', formData, { headers: { 'Content-Type': 'multipart/form-data' } })`), so the Authorization header, CSRF token, and `/g/{currentGroupSlug}` URL rewrite are applied consistently with every other upload in the app. The only thing overridden per-call is `Content-Type`, because the axios client defaults to `application/vnd.api+json` which is wrong for multipart.

Error handling is switched to the axios shape (`err.response?.data?.errors?.[0]?.detail`) to match the rest of the view.

File: go/apiserver/uploads.go [hunk 17]
```diff
@@ -122,9 +123,10 @@ func (api *uploadsAPI) handleImageUpload(w http.ResponseWriter, r *http.Request)
...
 	now := time.Now()
 	fileEntity := models.FileEntity{
-		TenantAwareEntityID: models.TenantAwareEntityID{
-			TenantID: user.TenantID,
-			UserID:   user.ID,
+		TenantGroupAwareEntityID: models.TenantGroupAwareEntityID{
+			TenantID:        user.TenantID,
+			GroupID:         appctx.GroupIDFromContext(r.Context()),
+			CreatedByUserID: user.ID,
 		},
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/uploads.go:136]: Good catch — fixed in 8ae91a7. All 6 upload handlers (`handleImageUpload`, `handleManualUpload`, `handleInvoiceUpload`, `handleFileUpload`, `handleLocationImageUpload`, `handleLocationFileUpload`) now reject with 500 when `appctx.GroupIDFromContext` returns `""` instead of silently persisting `group_id = ""`. Under the current routing this can only fire if someone mounts an upload handler outside the `/api/v1/g/{groupSlug}/...` tree, but the guard makes that failure mode loud rather than silent.

File: go/models/group_membership.go [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+type GroupMembership struct {
+	//migrator:embedded mode="inline"
+	TenantOnlyEntityID
+
+	// GroupID references the location group this membership belongs to.
+	//migrator:schema:field name="group_id" type="TEXT" not_null="true" foreign="location_groups(id)" foreign_key_name="fk_membership_group"
+	GroupID string `json:"group_id" db:"group_id"`
+
+	// MemberUserID references the user who is a member of the group.
+	// Named MemberUserID (not UserID) to avoid collision with TenantAwareEntityID.UserID.
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_membership.go]: Good catch — the comment was left over from before the struct switched to `TenantOnlyEntityID`. Updated in 8ae91a7 to state that `MemberUserID` is the *subject* of the membership row (the user being admitted to the group), not metadata about who created it, and to point at the `WithCreatedByUserID` pattern for creator-of-record fields on other entities. The stale `TenantAwareEntityID.UserID` reference is gone.

File: go/models/group_invite.go [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+	fields = append(fields,
+		validation.Field(&gi.TenantID, rules.NotEmpty),
+		validation.Field(&gi.GroupID, rules.NotEmpty),
+		validation.Field(&gi.Token, rules.NotEmpty),
+		validation.Field(&gi.CreatedBy, rules.NotEmpty),
+		validation.Field(&gi.ExpiresAt, validation.Required),
+	)
+
+	return validation.ValidateStructWithContext(ctx, gi, fields...)
+}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/group_invite.go:108]: Fixed in 8ae91a7. `ValidateWithContext` now rejects `ExpiresAt` values that would be born already expired:

```go
validation.Field(&gi.ExpiresAt,
    validation.Required,
    validation.Min(time.Now()).Error("expires_at must be in the future"),
),
```

`validation.Min(time.Now())` rejects both the zero time and any past time, so an invite can't be created in an already-invalid state.

File: go/apiserver/group_slug_resolver_test.go [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+
+	c.Assert(rr.Code, qt.Equals, http.StatusNotFound)
+}
+
+// TestGroupSlugResolver_NonMember asserts that a slug that resolves to an
+// active group, but where the requesting user is NOT a member, is rejected
+// with 403 (Forbidden) rather than 404. This is the cross-group isolation
+// contract: existence is not leaked to non-members, but we still distinguish
+// "you are not a member" from "that group does not exist" based on whether
+// GetGroupBySlug returned a row — and here it did.
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/group_slug_resolver_test.go]: You're right — the original comment hand-waved the information-disclosure question. I've rewritten it in 8ae91a7 to make the trade-off explicit rather than dismiss it. The expanded comment now spells out:

1. Yes, returning 403 for an existing-but-not-joined group *does* leak group existence to an authenticated user in the same tenant.
2. Slugs are 22+ chars of base64url random, so a non-member typically only learns a slug because someone shared the invite link. Distinguishing "no such group" from "you're not in it" is a real UX win — the user knows to ask for an invite rather than staring at a generic 404.
3. Invite IDs (the DELETE path) are different: they're DB-internal, an admin probing them gets no legitimate UX benefit, and that's specifically why that path uses `NewMaskedNotFoundError` / `services.ErrInviteNotInGroup`.
4. The comment now flags that if the product decision changes (e.g. slugs become guessable or user-picked), flip this to a masked 404 and update the test.

[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/group_slug_resolver_test.go]: Fixed in 8ae91a7 — see my reply to the earlier duplicate comment on this file. The rewritten comment no longer hand-waves the information-disclosure question; it explicitly acknowledges that 403 leaks group existence, justifies it (22+ char base64url random slugs → a non-member learns a slug only via a shared invite link → "ask for an invite" is a better UX than a masked 404), contrasts it with the invite DELETE path where we *do* return a masked 404 for exactly the probing reason you raised, and flags that the trade-off needs re-evaluating if slugs ever become guessable.

File: go/models/user.go [hunk 21]
```diff
@@ -54,8 +35,6 @@ type User struct {
 	PasswordHash string `json:"-" db:"password_hash" userinput:"false"`
 	//migrator:schema:field name="name" type="TEXT" not_null="true"
 	Name string `json:"name" db:"name"`
-	//migrator:schema:field name="role" type="TEXT" not_null="true" default="user"
-	Role UserRole `json:"role" db:"role"`
 	//migrator:schema:field name="is_active" type="BOOLEAN" not_null="true" default="true"
 	IsActive bool `json:"is_active" db:"is_active"`
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/user.go:39]: Same root cause as the other PR-description comment — the description was written when the group models still embedded `TenantAwareEntityID` and was left stale after the switch to `TenantOnlyEntityID`. I've rewritten the PR description to match the code: tenant-only group metadata, migration drops the inherited `user_id` columns, and `TenantGroupAwareEntityID` is defined but unused until Phase 6.

File: go/models/location_group_test.go [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+			name: "name too long",
+			group: models.LocationGroup{
+				TenantOnlyEntityID: models.TenantOnlyEntityID{TenantID: "tenant-1"},
+				Slug:               validSlug,
+				Name:               string(make([]byte, 101)),
+				Status:             models.LocationGroupStatusActive,
+				CreatedBy:          "user-1",
+			},
+			wantErr: true,
+		},
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/models/location_group_test.go:125]: Fixed in 8ae91a7. Replaced `string(make([]byte, 101))` (101 NUL bytes — legal for the length check but unrepresentative input) with `strings.Repeat("a", 101)`. Easier to read and what most readers would assume the test is actually checking.

File: go/registry/postgres/postgres.go [hunk 23]
```diff
@@ -48,19 +48,35 @@ func NewFactorySet(dbx *sqlx.DB) *registry.FactorySet {
...
 	})
 
+	if groupID != "" {
+		ctx = appctx.WithGroup(ctx, &models.LocationGroup{
+			TenantOnlyEntityID: models.TenantOnlyEntityID{
+				EntityID: models.EntityID{ID: groupID},
+				TenantID: tenantID,
+			},
+		})
+	}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/registry/postgres/postgres.go]: Fixed in 8ae91a7. Removed `NewRegistrySetWithUserID` from both the postgres and memory backends. `NewRegistrySetWithUserAndGroupID` (postgres) is now the only entry point for building a per-request registry set directly, and its doc comment states that group-scoped data registries require a non-empty `groupID` — so the silent "works for tenant-only tables, breaks for group-scoped tables" foot-gun is gone. Verified there are no external callers before deleting.

File: frontend/src/services/api.ts [hunk 24]
```diff
@@ -88,6 +88,33 @@ api.interceptors.request.use(
...
+        ]
+        for (const prefix of groupScopedPrefixes) {
+          if (config.url.startsWith(prefix)) {
+            const suffix = config.url.slice('/api/v1'.length)
+            config.url = `/api/v1/g/${groupSlug}${suffix}`
+            break
+          }
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/services/api.ts:123]: Fixed in 8ae91a7. The interceptor now uses `encodeURIComponent(groupSlug)`:

```ts
config.url = `/api/v1/g/${encodeURIComponent(groupSlug)}${suffix}`
```

You're right that today's slugs (22+ chars of base64url) are URL-safe by construction, so this is effectively a no-op for the current format. But the server side owns the slug format, and if it ever changes (user-picked names, shorter tokens, different alphabet) this interceptor keeps working without needing a coordinated frontend change. Cheap insurance.
